website/statistics.py

```diff
@@ -267,10 +268,14 @@ def is_program_modified(self, program, full_adventures, teacher_adventures):
         can_save = True
         for snippet in adventure_snippets:
             seq_match = SequenceMatcher(None, snippet, student_code)
-            if seq_match.ratio() > 0.95:
+            # Allowing a difference of more than 10% or the student filled the placeholders
+            if seq_match.ratio() > 0.95 or (not self.has_placeholder(student_code) and self.has_placeholder(snippet)):
```

Comment:
i tested it with the restaurant adventure in level 1, and i can't seem the sutdent's progress still! The function is returning False for both snippets!

Reply:
You're right!! It should work now!

Reply:
Cool and it does:)